cmd/hubagent/workload/setup.go

```diff
@@ -220,18 +230,66 @@ func SetupControllers(ctx context.Context, wg *sync.WaitGroup, mgr ctrl.Manager,
 			return err
 		}
 
-		// Set up a new controller to do rollout resources according to CRP rollout strategy
+		if opts.EnableResourcePlacement {
+			for _, gvk := range rpRequiredGVKs {
+				if err = utils.CheckCRDInstalled(discoverClient, gvk); err != nil {
+					klog.ErrorS(err, "unable to find the required CRD", "GVK", gvk)
+					return err
+				}
+			}
+			klog.Info("Setting up resourcePlacement controller")
+			resourcePlacementController = controller.NewController(rpControllerName, controller.NamespaceKeyFunc, crpc.Reconcile, rateLimiter)
+			klog.Info("Setting up resourcePlacement watcher")
+			if err := (&clusterresourceplacementwatcher.Reconciler{
+				PlacementController: resourcePlacementController,
+			}).SetupWithManagerForResourcePlacement(mgr); err != nil {
+				klog.ErrorS(err, "Unable to set up the resourcePlacement watcher")
+				return err
+			}
+
+			klog.Info("Setting up resourceBinding watcher")
+			if err := (&clusterresourcebindingwatcher.Reconciler{
+				PlacementController: resourcePlacementController,
+				Client:              mgr.GetClient(),
+			}).SetupWithManagerForResourceBinding(mgr); err != nil {
+				klog.ErrorS(err, "Unable to set up the resourceBinding watcher")
+				return err
+			}
+
+			klog.Info("Setting up schedulingPolicySnapshot watcher")
+			if err := (&clusterschedulingpolicysnapshot.Reconciler{
+				Client:              mgr.GetClient(),
+				PlacementController: resourcePlacementController,
+			}).SetupWithManagerForSchedulingPolicySnapshot(mgr); err != nil {
+				klog.ErrorS(err, "Unable to set up the schedulingPolicySnapshot watcher")
```

Comment:
since we reuse all those controllers, we probably want to rename them in the next PR.